nx/test/nx/serving_test.exs

```diff
@@ -631,15 +631,15 @@ defmodule Nx.ServingTest do
 
       assert_receive {:execute, 0, executor}
       send(serving_pid, {:system, {self(), make_ref()}, {:terminate, :shutdown}})
-      send(executor, :continue)
```

Comment:
The race condition here seems to be that before the serving GenServer enters `terminate/2`, the first execution would finish and the second would start.

To solve this, we let GenServer enter `terminate/2` trigger `:noproc` on the queued caller and only then finish the execution in progress.